src/parser/asts/variables.c

```
@@ -1,72 +1,85 @@
+    var->type = malloc(1);
+
+    switch(result.tokens[index].type) {
+        case TYPE_INT32:
+            var->type[0] = 0x01;
+            break;
+        case VAR:
+            var->type[0] = 0x00;
+            break;
+        default:
+            printf("Error: Disallowed token as variable type!\n");
+            return NULL;
```

cppcheck (2.10-2)

[error] Memory leak

(memleak)